tests/unit/customizations/wizard/test_app.py

```diff
@@ -1523,3 +1538,105 @@ def test_exception_handler(self):
         values = self.get_wizard_values({'use_handler': self.handler})
         value = values['handler_value']
         self.exception_handler.assert_called_with(e)
+
+
+class FakePathCompleter(PathCompleter):
+    def get_completions(self, document, complete_event):
+        prefix_len = len(document.text)
+
+        yield from [
+            Completion('file1'[prefix_len:], 0, display='file1'),
+            Completion('file2'[prefix_len:], 0, display='file2'),
+        ]
+
+
+class TestPromptCompletionWizardApplication(BaseWizardApplicationTest):
+    def setUp(self):
+        self.patch = mock.patch(
```

Comment:
Instead of patching, what is your thoughts on just using a temporary/sandboxed directory to work with? Specifically, we have the `FileCreator` abstraction that makes it easy to create temporary files: https://github.com/aws/aws-cli/blob/4c71115ef0538004ab0df93f55b92b5399667a94/awscli/testutils.py#L501 and we often use it in dealing with the filesystem layer.

Reply:
I had an issue with `FileCreator` clean up on windows containers so I'd prefer not to use it if it's not necessary and in this case by adding real files we will test not our code but `PromptToolkit` `PathCompleter`

Reply:
Yeah I think ideally the completer would be injectable, but that would require quite a bit of refactoring. I think the main reason that I brought up the suggestion was that I really do not like when we use patching because it has caused issues for us in the past related to patches not being properly cleaned up and also covering up bugs.

It's a little concerning that the `FileCreator` abstraction is not working properly, but I'm assuming this will get addressed in the other GitHub action PR you have. Otherwise, I'm fine keeping the test as is.